### PR TITLE
[Fix] Remove `fp16` folder in `configs`.

### PR DESCRIPTION
**Pull request by MengzhangLI**
Remove 4 FP16 config files into their corresponding folders.

### CODE REVIEW

File: .dev/batch_test_list.py [hunk 1]
```diff
@@ -116,8 +116,8 @@
 ]
 fp16 = [
     dict(
-        config='configs/fp16/deeplabv3plus_r101-d8_512x1024_80k_fp16_cityscapes.py', # noqa
-        checkpoint='deeplabv3plus_r101-d8_512x1024_80k_fp16_cityscapes-cc58bc8d.pth', # noqa
+        config='configs/deeplabv3plus/deeplabv3plus_r101-d8_fp16_512x1024_80k_cityscapes.py', # noqa
```
Comment (Junjun2016) [inline .dev/batch_test_list.py:119]: fp16

File: README.md [hunk 2]
```diff
@@ -73,7 +73,6 @@ Supported methods:
 - [x] [UNet (MICCAI'2016/Nat. Methods'2019)](configs/unet)
 - [x] [PSPNet (CVPR'2017)](configs/pspnet)
 - [x] [DeepLabV3 (ArXiv'2017)](configs/deeplabv3)
-- [x] [Mixed Precision (FP16) Training (ArXiv'2017)](configs/fp16)
 - [x] [BiSeNetV1 (ECCV'2018)](configs/bisenetv1)
```
Comment (Junjun2016) [inline README.md:76]: Can we add one item named `supported training tricks` in the future?

File: configs/deeplabv3/README.md [hunk 3]
```diff
@@ -39,6 +39,7 @@
...
+| DeepLabV3 (FP16)  | R-101-D8 | 512x1024  |   80000 | 5.75     | 3.86           | 80.48 |             - | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/deeplabv3/deeplabv3_r101-d8_fp16_512x1024_80k_cityscapes.py)     | [model](https://download.openmmlab.com/mmsegmentation/v0.5/deeplabv3/deeplabv3_r101-d8_fp16_512x1024_80k_cityscapes/deeplabv3_r101-d8_fp16_512x1024_80k_cityscapes_20200717_230920-774d9cec.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/deeplabv3/deeplabv3_r101-d8_fp16_512x1024_80k_cityscapes/deeplabv3_r101-d8_fp16_512x1024_80k_cityscapes_20200717_230920.log.json)                 |
```
Comment (Junjun2016) [inline configs/deeplabv3/README.md:42]: Add a note for `FP16` is better.

File: configs/deeplabv3/deeplabv3.yml [hunk 4]
```diff
@@ -157,6 +157,27 @@ Models:
...
+  Metadata:
+    backbone: R-101-D8
+    crop size: (512,1024)
+    lr schd: 80000
+    inference time (ms/im):
+    - value: 259.07
+      hardware: V100
+      backend: PyTorch
+      batch size: 1
+      mode: FP32
```
Comment (Junjun2016) [inline configs/deeplabv3/deeplabv3.yml:171]:
```suggestion
      mode: FP16
```

Comment (Junjun2016) [reply configs/deeplabv3/deeplabv3.yml:171]: Should update the yml2md hook.

File: configs/fcn/fcn.yml [hunk 5]
```diff
@@ -155,6 +155,27 @@ Models:
...
+  Metadata:
+    backbone: R-101-D8
+    crop size: (512,1024)
+    lr schd: 80000
+    inference time (ms/im):
+    - value: 115.74
+      hardware: V100
+      backend: PyTorch
+      batch size: 1
+      mode: FP32
```
Comment (Junjun2016) [inline configs/fcn/fcn.yml:169]: Same here

File: configs/pspnet/pspnet.yml [hunk 6]
```diff
@@ -158,6 +158,27 @@ Models:
...
+  Metadata:
+    backbone: R-101-D8
+    crop size: (512,1024)
+    lr schd: 80000
+    inference time (ms/im):
+    - value: 114.03
+      hardware: V100
+      backend: PyTorch
+      batch size: 1
+      mode: FP32
```
Comment (Junjun2016) [inline configs/pspnet/pspnet.yml:172]: Same here

File: docs/model_zoo.md [hunk 7]
```diff
@@ -129,7 +129,7 @@ Please refer to [CGNet](https://github.com/open-mmlab/mmsegmentation/blob/master
 
 ### Mixed Precision (FP16) Training
 
-Please refer [Mixed Precision (FP16) Training](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/fp16) for details.
+Please refer [Mixed Precision (FP16) Training] on BiSeNetV2 (https://github.com/open-mmlab/mmsegmentation/blob/master/configs/bisenetv2/bisenetv2_fcn_fp16_4x4_1024x1024_160k_cityscapes.py) for details.
```
Comment (Junjun2016) [inline docs/model_zoo.md:132]: It seems that some algorithms are missing in the model zoo, should add in the future.

**Comment by RockeyCoss** [comment]
Please merge the master branch into your branch, thank you.

**Review by Junjun2016 — APPROVED** [review]

**Pull request merged**